src/main/scala/extractors/Extractor.scala

```diff
@@ -0,0 +1,36 @@
+package extractors
+
+object Extractor {
+  def getDomainsInUa(domains: List[String]): List[String] = domains.filter(isDomainInUa)
+
+  private def isDomainInUa(domain: String): Boolean = {
+    domain match {
+      case Domain("ua", "in", _*) => true
+      case _ => false
+    }
+  }
+
+  def printEmailsInGmailDomain(emails: Array[String]): Unit = getEmailsInGmailDomain(emails).foreach(println)
+
+  private def getEmailsInGmailDomain(emails: Array[String]): Array[String] = emails.filter(isEmailInGmailDomain)
+
+  private def isEmailInGmailDomain(email: String): Boolean = {
+    email match {
+      case Email(_, Domain("com", "gmail")) => true
+      case _ => false
+    }
+  }
+}
+
+object Email {
+  def apply(localPart: String, domain: String): String = localPart + "@" + domain
+
+  def unapply(email: String): Option[(String, String)] = {
+    val parts = email.split("@")
+    if (parts.length == 2) Some(parts(0), parts(1)) else None
```

Comment:
Pattern matching will be more readable here:
```suggestion
      email.split("@") match {
          case Array(first, second) => Some(first, second)
          case _ => None
    }
```

Reply:
changed to pattern matching